tests/namespaces.bats

```diff
@@ -63,58 +63,37 @@ load helpers
   # Check that with settings that don't require a user namespace, we can request to use a per-container network namespace.
   run_buildah run $RUNOPTS --net=container "$ctr" readlink /proc/self/ns/net
   if [[ $output == $mynetns ]]; then
-      expect_output "[output should not be '$mynetns']"
+      die "[/proc/self/ns/net (--net=container) should not be '$mynetns']"
   fi
 
   run_buildah run $RUNOPTS --net=private "$ctr" readlink /proc/self/ns/net
   if [[ $output == $mynetns ]]; then
-      expect_output "[output should not be '$mynetns']"
+      die "[/proc/self/ns/net (--net=private) should not be '$mynetns']"
   fi
 }
 
-idmapping_check_namespace() {
-  local _uidmapargs=$1
-  local _gidmapargs=$2
-  local _mynamespace=$3
-  local _output=$4
+# Helper for idmapping test: check UID or GID mapping
+# NOTE SIDE EFFECT: sets $rootxid for possible use by caller
+idmapping_check_map() {
+  local _output_idmap=$1
+  local _expect_idmap=$2
+  local _testname=$3
 
-  [ "$_output" != "" ]
-  if [ -z "${_uidmapargs}${_gidmapargs}" ]; then
-    if test "$BUILDAH_ISOLATION" != "chroot" -a "$BUILDAH_ISOLATION" != "rootless" ; then
-      expect_output --from="$_output" "$_mynamespace"
-    fi
-  else
-    [ "$_output" != "$_mynamespace" ]
-  fi
-}
+  [ -n "$_output_idmap" ]
+  local _idmap=$(sed -E -e 's, +, ,g' -e 's,^ +,,g' <<< "${_output_idmap}")
+  expect_output --from="$_idmap" "${_expect_idmap}" "$_testname"
 
-idmapping_check_map() {
-  local _output_uidmap=$1
-  local _output_gidmap=$2
-  local _expect_uidmap=$3
-  local _expect_gidmap=$4
```

Comment:
Converted to check a generic X id map, to avoid the duplicate complicated `sed` lines. It's better IMO to have two calls to this function, one with UID and one with GID, than to duplicate the complicated `sed` stuff.